### PR TITLE
Robotics console clean-up

### PR DESCRIPTION
**Pull request by boy2mantwicethefam**
It looks a bit nicer.

- Removed self-destruct sequence from the robotics console. Nobody used it, it took 60 seconds to detonate borgs when you could do it manually
- Because there's no self-destruct sequence page, there is no reason for a main menu, so now the only screen is the cyborg status screen
- Malf AI can now see if their borg is already hacked and their emagged module is unlocked. They won't notice if the borg is a rogue borg though
- Robotics console now checks if the robot is on the same z-level as it, if it isn't it can't send commands.
- Added defines that will be used in the future for borgs' emagged status (ROBOT_UNHACKED, ROBOT_HACKED, ROBOT_MALFHACKED)
- A little bit of clean-up

:cl:
 * tweak: The robotics console looks a bit cleaner.
 * tweak: Malfunctioning AI can now see if their robot is hacked by them. Subverted and traitor robots are still indistinguishable however.
 * tweak: The robotics console now only works on robots that are on the same z-level as it.
 * rscdel: Removed the robotics console self-destruct sequence.

### CODE REVIEW

**Review by jknpj — CHANGES_REQUESTED** [review]
If you're going to make emagged not binary, change the whole code to use your defines. Also do NOT make such generic-named defines. Use ROBOT_HACKED, ROBOT_UNHACKED and ROBOT_MALFHACKED instead.

**Comment by SonixApache** [comment]
\>Can now only see borgs on the same Z-level
Please don't do this, it'll make it impossible for AIs (and others) to retrieve spaced borgs with telesci.

**Comment by boy2mantwicethefam** [comment]
@jknpj The rest will come later, but sure I can change the defines
@SonixApache Alright, but I'm gonna make it so that the robots show up in the console but can't have commands done to them if they're out of range

**Comment by SonixApache** [comment]
The whole point is to remotely detonate them so you can retrieve their GPS-enabled MMI.

**Comment by boy2mantwicethefam** [comment]
Why did I think they have coordinates attached to them? Should I make coordinates showing up in this PR or a separate PR?

**Comment by SonixApache** [comment]
Do it seperate, it'd probably get flak for powergay.

**Comment by boy2mantwicethefam** [comment]
I think I'll need this PR judged first because it'll also gonflict, don't forget to remind me please

File: code/game/machinery/computer/robot.dm [hunk 1]
```diff
@@ -8,98 +8,66 @@
...
 
 /obj/machinery/computer/robotics/attack_paw(var/mob/user as mob)
-
 	return src.attack_hand(user)
-	return
 
 /obj/machinery/computer/robotics/attack_hand(var/mob/user as mob)
 	if(..())
 		return
-	if (src.z > 6)
```
Comment (jknpj) [inline code/game/machinery/computer/robot.dm:34]: Why remove this?

**Comment by Intigracy** [comment]
```Removed self-destruct sequence from the robotics console. Nobody used it, it took 60 seconds to detonate borgs when you could do it manually```

Incorrect, self destructing the borgs via timer is the genuinely best way to blow all borgs as an antag since when the AI looks at the console you're 60 seconds gone already

**Comment by boy2mantwicethefam** [comment]
@Intigracy AI can switch it off because there's no check against it (though there is also no prompt about it either, so you can get away with it and the AI won't notice until it's too late), and I genuinely haven't heard of borgs being blown up the destruction sequence way in the last several months at least, AI is by itself unlikely to notice if its borgs are blown up manually until someone yells about robots being detonated cause MMIs can't talk over the radio by default, anyway self-destruct sequence is genuinely never used, that's why I assume it's safe to remove to clean up the robotics console code

**Comment by jknpj** [comment]
Intigracy makes a valid point.

**Comment by boy2mantwicethefam** [comment]
I know he does, I'm trying to argue that the removal of the self-destruct thing is okay because it's extremely rarely used

File: code/game/machinery/computer/robot.dm (hunk 1 above)
Comment (boy2mantwicethefam) [reply code/game/machinery/computer/robot.dm:34]: there's no real need for this check if borgs can't have things done to them if they're in separate z-levels. I can't imagine a scenario where the robotics console would end up in z-level 7 and then end up being problematic for station borgs, also considering that this change prevents that too.

File: code/game/machinery/computer/robot.dm [hunk 2]
```diff
@@ -110,65 +78,11 @@
...
 				if(R)
-					if(istype(usr, /mob/living/silicon/ai))
-						if (R.connected_ai != usr)
-							return
-					if(istype(usr, /mob/living/silicon/robot))
-						if (R != usr)
-							return
-					if(R.scrambledcodes)
-						return
+					sanity_check(usr, R)
```
Comment (MadmanMartian) [inline code/game/machinery/computer/robot.dm:85]: Fix this.

Comment (Kurfursten) [inline code/game/machinery/computer/robot.dm:85]: this needs to be in the if statement

**Comment by N3X15** [comment]
Screenshots, for one.

**Comment by PJB3005** [comment]
Make timed self destruct unable to be cancelled by AI so it isn't a potential newbie trap, and still a great antag tool (person gets told to blow borgs, hits timed destruct instead of manually destructing borgs, AI can just stop it)

Also if you're gonna go on about making the interface looking cleaner then at least port it to Nano or something.

**Comment by Kurfursten** [comment]
Zth said he would drop his complaints if you return the destruct-all button

**Comment by boy2mantwicethefam** [comment]
I will go back to this PR when I feel like it again, so I'm putting a reminder here that the self-destruct button should be a central red text above the borg text

**Comment by Kurfursten** [comment]
Tell us if you ever come back to this

**Comment by boy2mantwicethefam** [comment]
I should make this less unatomic

**Pull request closed without merging**